### PR TITLE
Server setup improvements and fixes.

### PR DESCRIPTION
**Pull request by Uberi**
- Document the pipeline a little bit from a bird's eye view.
- Fix a typo.
- Fix test error when encoding isn't specified and extended plane characters are used.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,21 @@ Code Overview
...
+
+1. Telemetry pings are generated by Firefox instances and are sent out from users' computers.
+    * To simulate this for testing purposes, run `python test/postGZ.py FOLDER_CONTAINING_PINGS` to submit all the pings in the specified folder.
+    * Sample pings can be found under one's Firefox profile, in the `saved-telemetry-pings` subfolder - they are human-readable JSON files.
+    * The `test/sample-saved_session-pings` folder can be used as an example: run `python test/postGZ.py test/test-saved_session-pings` while the Node server is running.
+2. The pings are received by a Node server (which can be found in `http/server.js`).
+    * Configure the server first by editing `http/server_config.json` - make sure to set the `log_path` (and ensure this folder exists) and `stats_log_file` options.
+    * To start the server locally, run `node server.js ./server_config.json`.
+    * The submitted pings are written to disk at the configured location.
+3. The server writes the pings to storage as fast as possible. Each ping written to disk is an LZMA compressed JSON file.
```
Comment (mreid-moz) [inline README.md]: This isn't quite true - pings are written to disk in batches (up to the configured max file size), and are written out in the same format they are submitted (either gzipped or raw, typically, but whatever bytes are sent are directly saved to these batch files).

File: README.md [hunk 2]
```diff
@@ -65,6 +65,21 @@ Code Overview
 
 These are the important parts of the Telemetry Server architecture.
 
+The simplified pipeline works as follows:
+
+1. Telemetry pings are generated by Firefox instances and are sent out from users' computers.
+    * To simulate this for testing purposes, run `python test/postGZ.py FOLDER_CONTAINING_PINGS` to submit all the pings in the specified folder.
+    * Sample pings can be found under one's Firefox profile, in the `saved-telemetry-pings` subfolder - they are human-readable JSON files.
+    * The `test/sample-saved_session-pings` folder can be used as an example: run `python test/postGZ.py test/test-saved_session-pings` while the Node server is running.
```
Comment (mreid-moz) [inline README.md]: Both `test/sample-saved_session-pings` and `test/postGZ.py` are missing from the PR.  Also, please use similar file naming to existing python code, so either "post_gz.py" or something even more descriptive like "submit_files.py"

File: process_incoming/process_incoming_standalone.py [hunk 3]
```diff
@@ -25,9 +25,9 @@
 from multiprocessing import Process, Queue
 from subprocess import Popen
 
-import boto.sqs
-from boto.exception import S3ResponseError
-from boto.s3.connection import S3Connection
+#import boto.sqs
+#from boto.exception import S3ResponseError
+#from boto.s3.connection import S3Connection
```
Comment (mreid-moz) [inline process_incoming/process_incoming_standalone.py]: Please undo this change - it will cause the code to fail in prodoction

File: process_incoming/process_incoming_standalone.py [hunk 4]
```diff
@@ -36,7 +36,7 @@
 from telemetry.util.compress import CompressedFile
 import telemetry.util.timer as timer
 import telemetry.util.files as fileutil
-import telemetry.util.s3 as s3util
+#import telemetry.util.s3 as s3util
```
Comment (mreid-moz) [inline process_incoming/process_incoming_standalone.py]: ditto

File: README.md [hunk 5]
```diff
@@ -65,6 +65,21 @@ Code Overview
...
+2. The pings are received by a Node server (which can be found in `http/server.js`).
+    * Configure the server first by editing `http/server_config.json` - make sure to set the `log_path` (and ensure this folder exists) and `stats_log_file` options.
+    * To start the server locally, run `node server.js ./server_config.json`.
+    * The submitted pings are written to disk at the configured location.
+3. The server writes the pings to storage as fast as possible. Each ping written to disk is an LZMA compressed JSON file.
+4. Via a queue, the incoming ping processor reads these compressed pings, converts them into a more compact format, then outputs them as LZMA compressed compacted JSON pings.
+    * To simulate this for testing purposes, simply run `test/convert.sh` to process all the pings (in the format generated by the Node server) in `test/work`, writing the results to `test/out`.
```
Comment (mreid-moz) [inline README.md]: `test/convert.sh` is not included in this PR.

File: README.md [hunk 6]
```diff
@@ -65,6 +65,21 @@ Code Overview
...
+    * To view the results in plaintext, run `test/unpack-compacted-pings.bash` in the same folder as the `out` directory. The human-readable outputs can be found in `test/test-ping-outputs`.
```
Comment (mreid-moz) [inline README.md]: `test/unpack-compacted-pings.bash` is not included.

**Comment by Uberi** [comment]
* Rename `unpack_compacted_pings.bash` to `unpack_compacted_pings.sh`.
* Revert changes to `server_config.spot.json`.
* Remove commented out code in `submit_pings.py`
* Clarify comment in `unpack_compacted_pings.sh`
